core/runtime/extern_asm.rs

```diff
@@ -0,0 +1,133 @@
+//! Build script module for compiling the external assembly (used for the entry
+//! point) and linking it into the process binary. Requires out_dir to be added
+//! to rustc's link search path.
+
+pub(crate) fn build_and_link(out_dir: &str) {
+    use std::env::var;
+    let arch = var("CARGO_CFG_TARGET_ARCH").expect("Unable to read CARGO_CFG_TARGET_ARCH");
+
+    // Identify the toolchain configurations to try for the target architecture.
+    // We support trying multiple toolchains because not all toolchains are
+    // available on every OS that we want to support development on.
+    let build_configs = match arch.as_str() {
+        "riscv32" => &[
+            // First try riscv64-unknown-elf, as it is the toolchain used by
+            // libtock-c.
+            AsmBuildConfig {
```

Comment:
Can we add a `riscv32-unknown-elf` without the `as_extra_args` as well

Reply:
Do you have a `riscv32-unknown-elf` toolchain? If yes then I'll probably put it first, as it seems to be the "correct" toolchain for the task. If not I'll put it last because it will be the only untested toolchain.

Reply:
Actually, if you have `riscv32-unknown-elf` maybe I should put it in the middle, because it still won't be tested by our CI.

Reply:
I have a `riscv32-oe-elf` toolchain, which is a vendored version of `riscv32-unknown-elf`. I don't think we should add that, but I suspect some distros will have a 32-bit only toolchain.

I would say put it in the middle.

Reply:
I added `riscv32-unknown-elf` between `riscv64-unknown-elf` and `riscv64-linux-gnu`.